jobQueue/jobWrapper.js

```diff
@@ -25,47 +28,127 @@ const jobQueue = require('./setup').jobQueue;
 const delayToRemoveJobs = 3000;
 
 /**
+ * Set log object params from job results.
+ * @param  {Object} jobResultObj - Job result object
+ * @param  {Object} logObject - Log object
+ */
+function mapJobResultsToLogObject(jobResultObj, logObject) {
+  if (jobResultObj.reqStartTime) {
+    logObject.totalTime = `${Date.now() - jobResultObj.reqStartTime}ms`;
+  }
+
+  if (jobResultObj.queueTime) {
+    logObject.queueTime = `${jobResultObj.queueTime}ms`;
+  }
+
+  if (jobResultObj.workTime) {
+    logObject.workTime = `${jobResultObj.workTime}ms`;
+  }
+
+  if (jobResultObj.dbTime) {
+    logObject.dbTime = `${jobResultObj.dbTime}ms`;
+  }
+
+  if (jobResultObj.recordCount) {
+    logObject.recordCount = jobResultObj.recordCount;
+  }
+
+  if (jobResultObj.errorCount) {
+    logObject.errorCount = jobResultObj.errorCount;
+  }
+}
+
+/**
  * Listen for a job completion event and clean up (remove) the job. The delay
- * is introduced to avoid the job.id leakage.
+ * is introduced to avoid the job.id leakage. If activity logs are enabled,
+ * update logObject and print log
  *
  * TODO: This needs to be moved to the clock process, once we start exposing
  * APIs to monitor the jobs.
  *
  * @param {Object} job - Job object to be cleaned up from the queue
  */
-function removeJobOnComplete(job) {
+function processJobOnComplete(job, logObject) {
   if (job) {
-    job.on('complete', () => {
+    job.on('complete', (jobResultObj) => {
       setTimeout(() => {
         job.remove();
       }, delayToRemoveJobs);
+
+      // if enableWorkerActivityLogs is enabled, update logObject
+      if (featureToggles.isFeatureEnabled('enableWorkerActivityLogs') &&
+       jobResultObj && logObject) {
+        mapJobResultsToLogObject(jobResultObj, logObject);
+
+        /* The second argument should match the activity type in
+         /config/activityLog.js */
+        activityLogUtil.printActivityLogString(logObject, 'worker');
+      }
     });
   }
 } // removeJobOnComplete
 
 /**
+ * Logs worker activity and remove job when complete
+ * @param  {Object} job - Worker job
+ * @param  {Object} req - Request object
+ */
+function logAndRemoveJobOnComplete(req, job) {
+  // if activity logs are enabled, log activity and remove job
+  if (featureToggles.isFeatureEnabled('enableWorkerActivityLogs')) {
```

Comment:
@pallavi2209 This function can be modified to be used with different activity logs ie. API, realtime, ... Not just worker activity. Once this pull request gets merged I might be able to modify it slightly for reuse.